presto-hive/src/main/java/com/facebook/presto/hive/HiveBasicStatistics.java

```diff
@@ -0,0 +1,144 @@
+import static com.google.common.base.Preconditions.checkArgument;
+import static java.util.Objects.requireNonNull;
+
+public class HiveBasicStatistics
+{
+    private static final String NUM_FILES = "numFiles";
+    private static final String NUM_ROWS = "numRows";
+    private static final String RAW_DATA_SIZE = "rawDataSize";
+    private static final String TOTAL_SIZE = "totalSize";
+
+    private final OptionalLong fileCount;
+    private final OptionalLong rowCount;
+    private final OptionalLong inMemoryDataSizeInBytes;
+    private final OptionalLong onDiskDataSizeInBytes;
+
+    public HiveBasicStatistics(
+            OptionalLong fileCount,
+            OptionalLong rowCount,
+            OptionalLong inMemoryDataSizeInBytes,
+            OptionalLong onDiskDataSizeInBytes)
+    {
+        this.fileCount = requireNonNull(fileCount, "fileCount is null");
+        fileCount.ifPresent(count -> checkArgument(count >= 0, "fileCount must be greater or equal than zero"));
+        this.rowCount = requireNonNull(rowCount, "rowCount is null");
+        rowCount.ifPresent(count -> checkArgument(count >= 0, "rowCount must be greater or equal than zero"));
+        this.inMemoryDataSizeInBytes = requireNonNull(inMemoryDataSizeInBytes, "inMemoryDataSizeInBytes is null");
+        inMemoryDataSizeInBytes.ifPresent(size -> checkArgument(size >= 0, "inMemoryDataSizeInBytes must be greater or equal than zero"));
+        this.onDiskDataSizeInBytes = requireNonNull(onDiskDataSizeInBytes, "onDiskDataSizeInBytes is null");
+        onDiskDataSizeInBytes.ifPresent(size -> checkArgument(size >= 0, "onDiskDataSizeInBytes must be greater or equal than zero"));
+    }
+
+    public OptionalLong getFileCount()
+    {
+        return fileCount;
+    }
+
+    public OptionalLong getRowCount()
+    {
+        return rowCount;
+    }
+
+    public OptionalLong getInMemoryDataSizeInBytes()
+    {
+        return inMemoryDataSizeInBytes;
+    }
+
+    public OptionalLong getOnDiskDataSizeInBytes()
+    {
+        return onDiskDataSizeInBytes;
+    }
+
+    @Override
+    public boolean equals(Object o)
+    {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+        HiveBasicStatistics that = (HiveBasicStatistics) o;
+        return Objects.equals(fileCount, that.fileCount) &&
+                Objects.equals(rowCount, that.rowCount) &&
+                Objects.equals(inMemoryDataSizeInBytes, that.inMemoryDataSizeInBytes) &&
+                Objects.equals(onDiskDataSizeInBytes, that.onDiskDataSizeInBytes);
+    }
+
+    @Override
+    public int hashCode()
+    {
+        return Objects.hash(fileCount, rowCount, inMemoryDataSizeInBytes, onDiskDataSizeInBytes);
+    }
+
+    @Override
+    public String toString()
+    {
+        return toStringHelper(this)
+                .add("fileCount", fileCount)
```

Comment:
it would be nice to skip fields that are absent (`OptionalLong.empty()`)

Reply:
`toString` is needed here rather for a debugging purposes. Missing some fields might be rather misleading (it may seem that those fields were just forgotten to be added to the toString output). When filed is nullable, we do print `null` in `toString`. I thing in general it should be similar for `Optional`. By `in general` i mean the cases when we don't need a specific `toString` (like in `Expression`), but we need a toString for a general debugging needs.

Reply:
fyi you can always `com.google.common.base.MoreObjects.ToStringHelper#omitNullValues` (just fyi, no change requested)